src/app/modules/auth/service/auth-guard.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { CanActivate } from '@angular/router';
+import { JwtHelperService } from '@auth0/angular-jwt';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuardService implements CanActivate{
+
+  constructor(private jwtHelper:JwtHelperService) { }
+
+  canActivate(): boolean {
+    const token:any = localStorage.getItem('token');
+
+    if (token==null || this.jwtHelper.isTokenExpired(token)){
+      return false;
+    }
+    
+    return true;
+  }
+}
```

Comment:
When i implemented login first time i said AuthGuard was implemented, that was RoleGuard which job is validating expected role of user. This is AuthGuard which is checking if user is logged in and if token is valid.